src/main/java/com/uber/jenkins/phabricator/CommentBuilder.java

```diff
@@ -123,7 +125,11 @@ public void addUserComment(String customComment) {
         if (hasComment()) {
             comment.append("\n\n");
         }
-        comment.append(String.format("```\n%s\n```\n\n", customComment));
+        if (preserveFormatting) {
+            comment.append(String.format("\n%s\n", customComment));
```

Comment:
I think the leading new line may be excessive here. The one below is to separate from the opening back ticks